### ISSUE TITLE
Storage node checkin not commited to the DB

### ISSUE THREAD

**Issue by littleskunk** [storj/storj#6788]
**Expected Behavior**
Once per hour, my storage nodes are checking in and signal that they are still online. The satellite always returns success on these but depending on my last_contact_success the satellite might not commit the new timestamp to the database if the old timestamp isn't older than 2 hours yet.

**Actual Behavior**
A few weeks ago I noticed that my node stopped getting any uploads for about 30 minutes or so. I still had download traffic in the logs so my node wasn't offline but the satellite counted my nodes as offline (last_contact_success older than 4 hours). The hourly checkins had some retries because of a different bug with DNS resolution (I will create another ticket for that) but in the end, the retry should have been successful. It took me some time to find out how the satellite could think my node would be offline while my node was working correctly.

Turns out we have a bug in the code: https://github.com/storj/storj/blob/af5d446107a3e188367413cf05d1db8704475282/satellite/overlay/service.go#L604-L610

In addition to the expected 2-hour window on top of my last_contact_success this code will also take the last_contact_failure into account when deciding if it should commit the new timestamp into the DB. So all it needs is some other process to set the last_contact_failure (still need to find out which process that is) and that's enough to let the satellite ignore the recent check-ins and the node counts as offline.

**Impact**
This might impact our upload performance in a few ways. I believe at any given time a small percentage of the nodes count as offline for a few minutes maybe up to an hour. Even on a bigger scale, this will be almost unnoticeable for the nodes itself and also not for us. But it will create additional repair traffic and also additional garbage collection. I believe the combination of all 3 side effects can have a noticeable effect on upload performance. The customer gets fewer nodes to upload to while these nodes have to deal with additional repair traffic and additional garbage collection.

**Comment by littleskunk** [comment]
I think I can fix this problem myself. Removing the last_contact_failure should do the trick. Stay tuned for a pull request.

**Comment by elek** [comment]
> So all it needs is some other process to set the last_contact_failure (still need to find out which process that is)

I don't understand this part. What kind of other process? I assume it's still your storagenode. It means that satellite failed to contact to your SN. 

I might be a real failure, which means the current code does make sense. 

Or did I miss sg?

**Comment by littleskunk** [comment]
I don't know yet which other process that is. It might be the repair upload. As explained there is another bug regarding DNS queries. The repair upload has the same problem and might be unwilling to contact my node even if it is online all the time. That might set the last_contact_failure. In any case that timestamp should never suppress a successful node checkin! That part of the equation is a bug for sure.

**Comment by littleskunk** [comment]
Oh no I made a mistake. There is an additional line that should still commit the timestamp into the DB: https://github.com/storj/storj/blob/af5d446107a3e188367413cf05d1db8704475282/satellite/overlay/service.go#L611

So back to the beginning and keep observing. All I have so far is a storage node log that shows my node is online all the time but was recognized as offline for a few minutes with 0 upload traffic.

**Comment by littleskunk** [comment]
And I made a second mistake with my observation. On my Grafana dashboard, I noticed a flat upload rate for some time while I was still seeing repair traffic in the logs the minute the upload traffic cut off. I thought this is because the satellite thought my node would be offline but it is the other way around. My ISP might have rotated my IP address. My router will update my dynDNS record. Customer uploads and download don't care until it is time for my next node checkin. Audit and repair have an additional fallback strategy and will perform an extra DNS query and contact me even before the next node checkin. That would also match what I am seeing in the logs and on the Grafana dashboard. Just for the record here is what caught my attention.

![image](https://github.com/storj/storj/assets/6333390/ed8261c2-2bbf-4df6-9511-0aca4b6b3f0a)

I would say for now issue closed since the current behavior seems to be correct. However, there was a time when the satellite was even sending a lot of downtime emails to me when it shouldn't. The DNS failure I talked about might have caused that. I will keep an eye on it and come back with a new bug report if it happens again.
